orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -85,6 +85,11 @@ def build_gradient_graph(accessor, user_args_requiring_grad, user_args_not_requi
     for argument_name in user_args_requiring_grad:
         all_args_requiring_gradient.append(argument_name)
 
+    # Run graph transformations to optimize the graph
+    model.CopyFrom(
```

Comment:
to confirm, this happens after our symbolic shape inferencing, right?

Reply:
We perform onnx shape inference before the graph transformations. We don't run the onnxruntime symbolic shape inference script though.